CMakeLists.txt

```diff
@@ -4,172 +4,127 @@ project(herb_description)
+set(ROBOTS_SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}/robots")
+set(BASE_OUTPUT_DIR "${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_SHARE_DESTINATION}")
+set(ROBOTS_OUTPUT_DIR "${BASE_OUTPUT_DIR}/robots")
+file(MAKE_DIRECTORY ${ROBOTS_OUTPUT_DIR})
+
+# Build component parts
+set(WAM_STANDALONE_URDF_XACRO "${ROBOTS_SRC_DIR}/wam_standalone.urdf.xacro")
+set(WAM_STANDALONE_URDF "${ROBOTS_OUTPUT_DIR}/wam.urdf")
+
+set(BH280_STANDALONE_URDF_XACRO "${ROBOTS_SRC_DIR}/bh280_standalone.urdf.xacro")
+set(BH280_STANDALONE_URDF "${ROBOTS_OUTPUT_DIR}/bh280.urdf")
+
+set(WAM_BH280_STANDALONE_URDF_XACRO "${ROBOTS_SRC_DIR}/wam_bh280_standalone.urdf.xacro")
+set(WAM_BH280_STANDALONE_URDF "${ROBOTS_OUTPUT_DIR}/wam_bh280.urdf")
+
+build_xacro(${WAM_STANDALONE_URDF_XACRO} ${WAM_STANDALONE_URDF} "")
+build_xacro(${BH280_STANDALONE_URDF_XACRO} ${BH280_STANDALONE_URDF} "")
+build_xacro(${WAM_BH280_STANDALONE_URDF_XACRO} ${WAM_BH280_STANDALONE_URDF} "")
+
+# HERB model matrix
+#
+# | LEFT  | RIGHT |
+# |-------|-------|
+# | BH280 | BH280 |
+# | BH280 | None  |
+# | None  | BH280 |
+# | None  | None  |
+
+# Default HERB (two BH280 hands)
+set(HERB_URDF_XACRO "${ROBOTS_SRC_DIR}/herb.urdf.xacro")
+set(HERB_URDF "${ROBOTS_OUTPUT_DIR}/herb.urdf")
+
+set(HERB_SRDF_XACRO "${ROBOTS_SRC_DIR}/herb.srdf.xacro")
+set(HERB_SRDF "${ROBOTS_OUTPUT_DIR}/herb.srdf")
+
+set(full_herb_args "left_bh280:=true" "right_bh280:=true")
+build_xacro(${HERB_URDF_XACRO} ${HERB_URDF} "${full_herb_args}")
+build_xacro(${HERB_SRDF_XACRO} ${HERB_SRDF} "")
+
+if(BUILD_ALL_HERB_MODELS)
+
+    # HERB with no right hand
+    set(HERB_NO_RIGHT_HAND_URDF "${ROBOTS_OUTPUT_DIR}/herb_bh280_left_no_right.urdf")
+    set(HERB_NO_RIGHT_HAND_SRDF "${ROBOTS_OUTPUT_DIR}/herb_bh280_left_no_right.srdf")
+
+    set(no_right_herb_args "left_bh280:=true" "right_bh280:=false")
+    build_xacro(${HERB_URDF_XACRO} ${HERB_NO_RIGHT_HAND_URDF} "${no_right_herb_args}")
+    build_xacro(${HERB_SRDF_XACRO} ${HERB_NO_RIGHT_HAND_SRDF} "right_hand:=false")
+
+    # HERB with no left hand
+    set(HERB_NO_LEFT_HAND_URDF "${ROBOTS_OUTPUT_DIR}/herb_no_left_bh280_right.urdf")
+    set(HERB_NO_LEFT_HAND_SRDF "${ROBOTS_OUTPUT_DIR}/herb_no_left_bh280_right.srdf")
+
+    set(no_left_herb_args "left_bh280:=false" "right_bh280:=true")
+    build_xacro(${HERB_URDF_XACRO} ${HERB_NO_LEFT_HAND_URDF} "${no_left_herb_args}")
+    build_xacro(${HERB_SRDF_XACRO} ${HERB_NO_LEFT_HAND_SRDF} "left_hand:=false")
+
+    # HERB with no hands
+    set(HERB_NO_HANDS_URDF "${ROBOTS_OUTPUT_DIR}/herb_no_left_no_right.urdf")
+    set(HERB_NO_HANDS_SRDF "${ROBOTS_OUTPUT_DIR}/herb_no_left_no_right.srdf")
+
+    set(no_right_herb_args "left_bh280:=false" "right_bh280:=false")
+    build_xacro(${HERB_URDF_XACRO} ${HERB_NO_HANDS_URDF} "${no_hands_herb_args}")
+    set(no_hands_herb_args "left_hand:=false" "right_hand:=false")
+    build_xacro(${HERB_SRDF_XACRO} ${HERB_NO_HANDS_SRDF} "${no_hands_herb_args}")
+
+endif(BUILD_ALL_HERB_MODELS)
```

Comment:
This will grow unwieldy if we have too many more options, e.g. add the BH-282. I am fine with leaving this as it is for now, but we should consider switching to nested `foreach` loops when that happens.